node/testing.go

```diff
@@ -27,3 +34,43 @@ func TestNode(t *testing.T, tp Type, opts ...Option) *Node {
 	require.NoError(t, err)
 	return nd
 }
+
+// WithFaultMaker is used to enable test cases with invalid blocks
+func WithFaultMaker() Option {
+	return func(sets *settings) {
+		sets.opts = append(sets.opts, fx.Replace(faultMaker))
+	}
+}
+
+func faultMaker(ctx context.Context,
+	b *types.Block,
+	comm *types.Commit,
+	vals *types.ValidatorSet,
+	dag format.NodeAdder) (*header.ExtendedHeader, error) {
+	log.Info("using fault header maker...")
```

Comment:
in the dynamic set of fault headers, we need more explicit info log (can be included in the issue creation described below 👇 in the next comment review line)

Reply:
added